test/fluree/db/query/misc_queries_test.clj

```diff
@@ -237,5 +237,13 @@
                   ["fluree:commit:sha256:bbstbkfuob2d73ubsc3xvqckbmn7z35esq5k7i3rolj7hed7li2ox"
                    :f/context
                    "fluree:memory://b6dcf8968183239ecc7a664025f247de5b7859ac18cdeaace89aafc421eeddee"]]
-                @(fluree/query db* {:select ['?s '?p '?o]
-                                    :where  [['?s '?p '?o]]}))))))))
+                 @(fluree/query db* {:select ['?s '?p '?o]
+                                     :where  [['?s '?p '?o]]})))))
+      (testing "Illegal reference queries"
+        (let [test-subject @(fluree/query db {:select ['?s '?p]
+                                              :where [['?s '?p 22]]})]
```

Comment:
For completion's sake, could we also have a test for the non-iri string object case that was in the bug report?